src/utils/postgrest-data-provider.js

```diff
@@ -0,0 +1,308 @@
+    requestHeaders.set("Authorization", options.user.token);
+  }
+
+  return fetch(url, { ...options, headers: requestHeaders })
+    .then(response =>
+      response.text().then(text => ({
+        status: response.status,
+        statusText: response.statusText,
+        headers: response.headers,
+        body: text
+      }))
+    )
+    .then(({ status, statusText, headers, body }) => {
+      let json;
+      try {
+        json = json2ParseBigint(body);
+      } catch (e) {
+        // not json, no big deal
+      }
+      if (status < 200 || status >= 300) {
+        return Promise.reject(new HttpError((json && json.message) || statusText, status, json));
+      }
+      return Promise.resolve({ status, headers, body, json });
+    });
+};
+
+/**
+ * Maps admin-on-rest queries to a postgrest API
+ *
+ * The REST dialect is similar to the one of FakeRest
+ * @see https://github.com/marmelab/FakeRest
+ * @example
+ * GET_MANY_REFERENCE
+ *              => GET http://my.api.url/posts/2
+ * GET_LIST     => GET http://my.api.url/posts?order=title.asc
+ * GET_ONE      => GET http://my.api.url/posts?id=eq.123
+ * GET_MANY     => GET http://my.api.url/posts?id=in.( 123,456,789 )
+ * UPDATE       => PATCH http://my.api.url/posts?id=eq.123
+ * CREATE       => POST http://my.api.url/posts
+ * DELETE       => DELETE http://my.api.url/posts?id=eq.123
+ */
+const postgrestClient = (apiUrl, httpClient = fetchJson) => {
+  const convertFilters = filters => {
+    const rest = {};
+
+    Object.keys(filters).map(function(key) {
+      switch (typeof filters[key]) {
+        case "string":
+          rest[key] = "ilike.*" + filters[key].replace(/:/, "") + "*";
+          break;
+
+        case "boolean":
+          rest[key] = "is." + filters[key];
+          break;
+
+        case "undefined":
+          rest[key] = "is.null";
+          break;
+
+        case "number":
+          rest[key] = "eq." + filters[key];
+          break;
+
+        case "object":
+          if (filters[key].constructor === Array) {
+            rest[key] = "cs.{" + filters[key].toString().replace(/:/, "") + "}";
+          } else {
+            Object.keys(filters[key]).map(val => (rest[`${key}->>${val}`] = `ilike.*${filters[key][val]}*`));
+          }
+          break;
+
+        default:
+          rest[key] = "ilike.*" + filters[key].toString().replace(/:/, "") + "*";
+          break;
+      }
+    });
+    return rest;
+  };
+
+  /**
+   * @param {String} type One of the constants appearing at the top if this file, e.g. 'UPDATE'
+   * @param {String} resource Name of the resource to fetch, e.g. 'posts'
+   * @param {Object} params The REST request params, depending on the type
+   * @returns {Object} { url, options } The HTTP request parameters
+   */
+  const convertRESTRequestToHTTP = (type, resource, params) => {
+    let url = "";
+    const options = {};
+    options.headers = new Headers();
+    const token = localStorage.getItem(localStorageKey);
```

Comment:
Can't we avoid using localStorage here? Seems pointless for a token that expires every 5 minutes.

Reply:
i'll confirm, i think you're right but we might end up seeing the login page anyway if we don't have that token there to be read immediately so the initial postgrest queries pass because of some react admin internals I didn't bother digging into yet. if it works without it i'll take it out otherwise i'll leave it, my plan was to try to polish up the login screen workflow at some point. (currently, you end up landing on a meaningless login screen when you first load the page)